chunky/src/java/se/llbit/chunky/renderer/scene/Sun.java

```diff
@@ -224,6 +196,11 @@ private void initSun() {
     emittance.set(color);
     emittance.scale(FastMath.pow(intensity, Scene.DEFAULT_GAMMA));
 
+    Sky sky = ((Scene) scene).sky();
+    if (sky.getSkyMode() == Sky.SkyMode.SIMULATED && sky.getSimulatedSky().needUpdate(this)) {
```

Comment:
As suggested above, you could just always call `sky.updateSimulatedSky(this)` if the implementation checks internally if an update is actually needed. That would simplify the caller side (and get rid of `needUpdate`)